src/sessions/PageManager.ts

```diff
@@ -17,11 +17,19 @@ export type Attributes = {
     parentPageId?: string;
     interaction?: number;
     pageTags?: string[];
+    // The value types of custom attributes are restricted to the types: string | number | boolean
+    // However, given that pageTags is a string array, we need to include it as a valid type
+    // Events will be verified by our service to validate attribute value types where
+    //  1) pageTags attribute value must be of type string[]
+    //  2) any other attribute value must be of the following types:
+    //      string | number | boolean
+    [k: string]: string | number | boolean | string[];
```

Comment:
thought: Thanks for documenting this. Maybe in the future we will support arrays in the general case as well :-).

Reply:
Yes, additional attributes can only be primitives but if we  remove that from line 26, `pageTags` can't be a `string[]` because of the type collision. Metadata schema will validate attributes where we drop any attributes with `string[]` type if it's not the `pageTags` attribute.